quipucords/tests/integration/test_smoker.py

```diff
@@ -0,0 +1,230 @@
+
+    @pytest.fixture
+    def scan_response(self, django_client, scan_id, scan_manager):
+        """Start a scan job and poll its results endpoint until completion."""
+        create_scan_job_response = django_client.post(f"scans/{scan_id}/jobs/")
+        assert create_scan_job_response.ok, create_scan_job_response.text
+        scan_manager.work()
+        response = django_client.get(f"scans/{scan_id}/")
+        attempts = 1
+        assert response.ok, response.text
+
+        completed_status = {ScanTask.COMPLETED, ScanTask.CANCELED, ScanTask.FAILED}
+        while (
+            scan_status := response.json()["most_recent"]["status"]
+        ) not in completed_status and attempts < self.MAX_RETRIES:
+            attempts += 1
+            backoff = 2**attempts
+            sleep(backoff)
+            response = django_client.get(f"scans/{scan_id}/")
+            assert response.ok, response.text
+
+        assert scan_status == ScanTask.COMPLETED, response.json()
+
+        return response
+
+    @pytest.fixture
+    def report_id(self, scan_response):
+        """Return the latest report id from performed scan."""
+        return scan_response.json()["most_recent"]["report_id"]
+
+    def test_details_report(
+        self,
+        django_client,
+        report_id,
+        expected_facts,
+    ):
+        """Sanity check details report."""
+        response = django_client.get(f"reports/{report_id}/details/")
+        assert response.ok, response.text
+        report_details_dict = response.json()
+        expected_details_report = {
+            "report_id": report_id,
+            "report_platform_id": mock.ANY,
+            "report_type": "details",
+            "report_version": mock.ANY,
+            SOURCES_KEY: [
+                {
+                    "facts": mock.ANY,
+                    "report_version": mock.ANY,
+                    "server_id": mock.ANY,
+                    "source_name": self.SOURCE_NAME,
+                    "source_type": self.SOURCE_TYPE,
+                }
+            ],
+        }
+        assert report_details_dict == expected_details_report
+        report_details_facts = report_details_dict[SOURCES_KEY][0]["facts"]
+        assert report_details_facts == expected_facts
+
+    @pytest.fixture
+    def expected_fingerprint_metadata(self, fingerprint_fact_map):
+        """
+        Return expected metadata dictionary.
+
+        Dict contained within deployments report > system fingerprints.
+        """
+        metadata = {}
+        for fingerprint_fact, raw_fact in fingerprint_fact_map.items():
+            metadata[fingerprint_fact] = {
+                "server_id": mock.ANY,
+                "source_name": self.SOURCE_NAME,
+                "source_type": self.SOURCE_TYPE,
+                "raw_fact_key": RawFactComparator(raw_fact),
+                "has_sudo": False,
+            }
+        return metadata
+
+    def test_deployments_report(  # pylint: disable=too-many-arguments
```

Comment:
I also ran into `too-many-arguments` recently, and I wonder if we disagree enough with pylint's "5 or fewer" opinion to add it to our global ignore list. At last check, we disable that check in 23 places (24 with this addition).

Since we do thorough code reviews, _hopefully_ we would notice and stop each other from needlessly tacking on lots of function arguments. Thoughts?

We don't need to act on this now. Seeing your comment just reminded me. 🙂

Reply:
I think it could have an exception for testing only  - kinda hard to avoid the fixtures XD - and/or configure pylint to support a higher value on this check. Outside of tests I personally think this number is a good advice, but that's just me.

Reply:
I don't _want_ to discourage GitHub issues because honestly they're so much better than Jira, but that issue belongs in Jira. 😕 Issues are going to slip through and be overlooked and forgotten if we have to remember to look in multiple places for them, and like it or not, Jira is the designated tracker for our project. I completely stopped looking at GitHub issues several weeks ago when I realized we were using Jira for everything (most things?).